python_on_whales/components/container/cli_wrapper.py

```diff
@@ -353,9 +370,38 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.remove = self.remove
 
-    def attach(self):
-        """Not yet implemented"""
-        raise NotImplementedError
+    def attach(
+        self,
+        container: ValidContainer,
+        detach_keys: Optional[str] = None,
+        no_stdin: bool = False,
+        sig_proxy: bool = True,
+    ) -> Union[None, str, Iterable[Tuple[str, bytes]]]:
```

Comment:
In which case does it return None and in which case does it returns an iterable for the logs?

If it's possible to return an iterable for the logs, would it be possible to test this in the unit tests without any mocks?

Reply:
Hey hi!
Glad to hear from you again :)

Whoops... basically during development I copied existing function headers as a template and did not pay much attention to their return types. Per my understanding,  _'attach' should always return None_ as it only attaches itself to an existing container and does not change/replace it in any way. Different types are instead returned by the 'run' function that launched the container in the first place.

I updated the required functions and their docstrings to reflect this.
No change was made to the existing unit tests.